kvwrapper/kvwrapper.go

```diff
@@ -37,20 +38,29 @@ func NewKVWrapperWithAuth(servers []string, wrapper KVWrapper, username, passwor
 	kvw := wrapper.NewKVWrapper(servers, username, password)
 	return kvw
 }
+
 func NewKVWrapper(servers []string, wrapper KVWrapper) KVWrapper {
 	kvw := wrapper.NewKVWrapper(servers, "", "")
 	return kvw
 }
 
 type KVFaker struct {
-	c map[string][]*KeyValue
+	c     map[string][]*KeyValue
+	mutex *sync.Mutex
 }
 
 func (f KVFaker) NewKVWrapper(servers []string, username, password string) KVWrapper {
 	f.c = make(map[string][]*KeyValue)
+	f.mutex = &sync.Mutex{}
 	return f
 }
+
 func (f KVFaker) Set(key string, val string, ttl uint64) error {
+	f.mutex.Lock()
+	defer func() {
```

Comment:
Why does this need to be wrapped in a function instead of just  `defer f.mutex.Unlock()`?

Reply:
Hmmm, good point. Updated (b89e460). I actually copied this from FD code, so we can probably change it in a bunch of places in FD as well.

Reply:
Updated in FD with https://git.corp.adobe.com/adobe-platform/flight-director/pull/434